src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -32,22 +32,26 @@ def __init__(self, config):
         self.condorAPI = PyCondorAPI()
         self.agentConfig = {}
         self.validSpeedDrainConfigKeys = ['CondorPriority', 'NoJobRetries', 'EnableAllSites']
-
         self.reqAuxDB = ReqMgrAux(self.config.General.ReqMgr2ServiceURL)
+        self.emailAlert = EmailAlert(config)
 
     @timeFunction
     def algorithm(self, parameters):
         """
         Update drainStats if agent is in drain mode
         """
         logging.info("Running agent drain algorithm...")
+        previousConfig = self.agentConfig
```

Comment:
This is doing a reference copy, not a value copy. You might want to use something like `copy.deepcopy` for this assignment such that a new dict object gets created.